### PR TITLE
[PDI-17005] Issues with Carte Jobs API executeJob endpoint in Pentaho…

### PR DESCRIPTION
**Pull request by LeonardoCoelho71950**
… REST API

@bantonio82 identified an error while trying to connect into a repository, via PDI.
That was because I added jaxws-rt in kettle-engine pom. But in compile scope, not in provided scope. Thus, we didn't have jaxws-rt on runtime.   
Nevertheless, I created a KettleClientTransportException. This way, we don't need to have jaxws-rt in kettle-engine runtime.  

@smmribeiro  Can you review this please, since you helped me on that? :)
@pentaho/tatooine @pentaho-lmartins @bantonio82

### CODE REVIEW

**Comment by LeonardoCoelho71950** [comment]
@graimundo I made some changes here.
Now we only use ClientTransportException at UnifiedRepositoryInvocationHandler.java.
Over there, if we notice a ClientTransportException, we will convert it into KettleAuthenticationException.

I couldn't simple used javax.xml.ws.WebServiceException (as I was thinking this morning), because we wanted to differentiate between an authentication error, due to bad credentials, and other errors like trying to connect to a BA server.

File: engine/src/main/java/org/pentaho/di/repository/KettleRepositoryNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+package org.pentaho.di.repository;
+
+import org.pentaho.di.core.exception.KettleException;
+
+/**
+ * This exception is thrown when there was an authentication error, while connecting into a repository.
```
Comment (graimundo) [inline engine/src/main/java/org/pentaho/di/repository/KettleRepositoryNotFoundException.java]: Description does not match exception name. Copy paste error?

File: engine/src/main/java/org/pentaho/di/www/ExecuteJobServlet.java [hunk 2]
```diff
@@ -202,26 +204,29 @@ public void doGet( HttpServletRequest request, HttpServletResponse response ) th
...
-        // Something unexpected occurred.
-        response.setStatus( HttpServletResponse.SC_INTERNAL_SERVER_ERROR );
-        message = BaseMessages.getString(
-          PKG, "ExecuteJobServlet.Error.UnexpectedError", Const.CR + Const.getStackTracker( ke ) );
-        out.println( new WebResult( WebResult.STRING_ERROR, message ) );
+      out.println( new WebResult( WebResult.STRING_ERROR, message ) );
+      return;
+    } catch ( KettleException ke ) {
+      // Authentication Error.
+      if (  ke.getCause() instanceof ExecutionException ) {
```
Comment (graimundo) [inline engine/src/main/java/org/pentaho/di/www/ExecuteJobServlet.java]: Are there tests for these cases?

File: plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/UnifiedRepositoryInvocationHandler.java [hunk 3]
```diff
@@ -58,6 +64,18 @@ private boolean lookupConnectException( Throwable root ) {
     return false;
   }
 
+  private boolean lookupAuthenticationException( Throwable root ) {
+    while ( root != null ) {
+      if ( root instanceof ClientTransportException ) {
```
Comment (graimundo) [inline plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/UnifiedRepositoryInvocationHandler.java]: Where are the tests for this case?

File: engine/src/main/java/org/pentaho/di/repository/KettleRepositoryNotFoundException.java (hunk 1 above)
Comment (LeonardoCoelho71950) [reply engine/src/main/java/org/pentaho/di/repository/KettleRepositoryNotFoundException.java]: Updated! *facepalm*

File: engine/src/main/java/org/pentaho/di/www/ExecuteJobServlet.java (hunk 2 above)
Comment (LeonardoCoelho71950) [reply engine/src/main/java/org/pentaho/di/www/ExecuteJobServlet.java]: There is a test to validate response error message, when a KettleAuthenticationException is thrown while connecting into a server.
https://github.com/pentaho/pentaho-kettle/blob/146e2160759081d40b451a35a11d953188a72e6f/engine/src/test/java/org/pentaho/di/www/ExecuteJobServletTest.java#L206

File: plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/UnifiedRepositoryInvocationHandler.java (hunk 3 above)
Comment (LeonardoCoelho71950) [reply plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/UnifiedRepositoryInvocationHandler.java]: Done

**Pull request merged**